docs/getting-started.md

```diff
@@ -0,0 +1,288 @@
+
+``` bash
+# Replace with your own public key
+aws ec2 import-key-pair --key-name cluster-api-provider-aws.sigs.k8s.io \
+  --public-key-material $(cat ~/.ssh/id_rsa.pub)
+```
+> Note: Only RSA keys are supported by AWS.
+
+##### PowerShell
+``` powershell
+$publicKey = [System.Convert]::ToBase64String( `
+  [System.Text.Encoding]::UTF8.GetBytes(((get-content ~/.ssh/id_rsa.pub))))
+Import-EC2KeyPair -KeyName cluster-api-provider-aws.sigs.k8s.io -PublicKeyMaterial $publicKey
+```
+
+### Setting up Minikube
+
+Minikube needs to be installed on your local machine, as this is what will
+be used by the Cluster API to bootstrap your cluster in AWS.
+
+[Instructions for setting up minikube][minikube] are on the Kubernetes website.
+
+#### Customising for Cluster API
+
+At present, the Cluster API provider runs minikube to create a new instance,
+but requires Kubernetes 1.9 and the kubeadm bootstrap method to work properly,
+so we configure Minikube as follows:
+
+``` bash
+minikube config set kubernetes-version v1.9.4
+minikube config set bootstrapper kubeadm
+```
+
+You should also select the hypervisor that works best for your environment.
+
+People have had the success with the following:
+
+* OS X: `minikube config set vm-driver hyperkit`
+* Linux: `minikube config set vm-driver kvm2`
+
+### Setting up the environment
+
+The current iteration of the Cluster API Provider AWS relies on credentials
+being present in your environment. These then get written into the cluster
+manifests for use by the controllers.
+
+You therefore need to set the following environment variables:
+
+##### Bourne shell
+
+``` bash
+# The region you will be running the cluster components in
+export AWS_REGION=us-east-1
+# The AWS credentials for bootstrap
+export AWS_ACCESS_KEY_ID="AKIAIOSFODNN7EXAMPLE"
+export AWS_SECRET_ACCESS_KEY="wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
+# The name of the SSH key stored in the EC2 API
+export SSH_KEY_NAME="cluster-api-provider-aws.sigs.k8s.io"
+```
+
+##### PowerShell
+``` powershell
+$ENV:AWS_REGION = "us-east-1"
+$ENV:AWS_ACCESS_KEY_ID="AKIAIOSFODNN7EXAMPLE"
+$ENV:AWS_SECRET_ACCESS_KEY="wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY"
+$ENV:SSH_KEY_NAME="cluster-api-provider-aws.sigs.k8s.io"
+```
+
+If you applied the bootstrap AWS CloudFormation, then you can use the IAM user
+that it created to scope down the permissions:
+
+##### Bourne shell
+``` shell
+export AWS_CREDENTIALS=$(aws iam create-access-key \
+  --user-name bootstrapper.cluster-api-provider-aws.sigs.k8s.io)
+export AWS_ACCESS_KEY_ID=$(echo $AWS_CREDENTIALS | jq .AccessKey.AccessKeyId -r)
+export AWS_SECRET_ACCESS_KEY=$(echo $AWS_CREDENTIALS | jq .AccessKey.SecretAccessKey -r)
+```
+
+##### PowerShell
+``` powershell
+$awsCredentials = New-IAMAccessKey -UserName bootstrapper.cluster-api-provider-aws.sigs.k8s.io
+$ENV:AWS_ACCESS_KEY_ID=$awsCredentials.AccessKeyId
+$ENV:AWS_SECRET_ACCESS_KEY=$awsCredentials.SecretAccessKey
+```
+
+### Generating cluster manifests
```

Comment:
This is probably where the "deploying a cluster" section should start.

Reply:
Done